### PR TITLE
Simplify `SortedRangeSet#compareTo` and remove  redundant casts

### PR DESCRIPTION
**Pull request by tangjiangling**
## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

File: core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java [hunk 1]
```diff
@@ -1104,17 +1104,11 @@ public Range toRange()
         @Override
         public int compareTo(RangeView that)
         {
-            int compare;
-            compare = compareLowBound(that);
+            int compare = compareLowBound(that);
```
Comment (findepi) [inline core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java:1107]: now the variable hold only comparison result for the low bound only, so could be renamed

Comment (tangjiangling) [reply core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java:1107]: How about renaming it to `lowBoundCompare` for readability?

File: core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java [hunk 2]
```diff
@@ -1104,17 +1104,11 @@ public Range toRange()
         @Override
         public int compareTo(RangeView that)
         {
-            int compare;
-            compare = compareLowBound(that);
```
[the remark this answers is not in the document]
Comment (tangjiangling) [reply core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java:1108]: Why? (or Does this change make sense to you?)

File: core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java (hunk 1 above)
Comment (tangjiangling) [reply core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java:1107]: Renamed in #13811

Comment (martint) [reply core/trino-spi/src/main/java/io/trino/spi/predicate/SortedRangeSet.java:1107]: I don’t think the existing name is less readable. If anything, I’d rename it to “comparison”. The fact that it’s the comparison for the lower bound is pretty obvious, as it’s the only thing being compared. There’s no need to make the name overly verbose — verbosity does not necessarily improve readability.

**Pull request merged**